SYCL/ESIMD/api/functional/ctors/ctor_vector_core.cpp

```diff
@@ -33,8 +33,7 @@ int main(int, char **) {
   sycl::device device = queue.get_device();
   // verify aspect::fp16 due to using sycl::half data type
   // verify aspect::fp64 due to using double data type
-  if (!device.is_host() && !device.has(sycl::aspect::fp16) &&
-      !device.has(sycl::aspect::fp64)) {
+  if (!device.has(sycl::aspect::fp16) && !device.has(sycl::aspect::fp64)) {
```

Comment:
The fix is correct.
But the underlying code (that is being fixed here) is totally incorrect IMO.
1) The test ctor_vector_core.cpp tests "core" types, i.e. all types except specific. "core" does not include double/half.
    So, this check must be not in this file, but in ctor_vector_fp_extra.cpp.
2) The check must be 'if (!has_fp16 || !has_fp64)' instead of '(!has_fp16 && !has_fp64).

There are several other tests like that.
This can be fixed in a separate PR (I can do that).